### PR TITLE
Revert self-hosted AppID validation and only disallow dots

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

This PR reverts the logic introduced in #5556 to validate self-hosted app-ids with RFC-1123. It only throws an error if the app-id contains a period (`.`).

A major reason is that upper-case in app-ids was no longer allowed after #5556, which broke the case-sensitive implementations in state store (app-id is prefixed) and pub/sub (consumer groups use app-id).

~~From #5556, we introduced stricter rules for appID validation in self-hosted mode. This PR relaxes the case restrictions by transforming the appID into a lowercase string. This will prevent breaking existing Dapr integrations where users are using appIDs with uppercase characters.~~

## Issue reference

Please reference the issue this PR will close: #5552

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shubham1172** [comment]
FYI @ItalyPaleAle @mukundansundar @artursouza

**Review by ItalyPaleAle — APPROVED** [review]

File: pkg/runtime/cli.go [hunk 1]
```diff
@@ -125,8 +125,16 @@ func FromFlags() (*DaprRuntime, error) {
 		os.Exit(0)
 	}
 
-	if err := validation.ValidateSelfHostedAppID(*appID); err != nil {
-		return nil, err
+	if *mode == string(modes.KubernetesMode) {
+		if err := validation.ValidateKubernetesAppID(*appID); err != nil {
+			return nil, err
```
Comment (mukundansundar) [inline pkg/runtime/cli.go:130]: this call is not needed here ... it is done during pod injection.

**Comment by yaron2** [comment]
I have concerns this might break subscriptions due to case sensitive pub/sub implementations. Need to look into this further.

File: pkg/runtime/cli.go [hunk 2]
```diff
@@ -125,8 +125,16 @@ func FromFlags() (*DaprRuntime, error) {
...
+		}
+	} else {
+		// Convert appID to lowercase to be compatible with the validation.
+		*appID = strings.ToLower(*appID)
+		if err := validation.ValidateSelfHostedAppID(*appID); err != nil {
+			return nil, err
```
Comment (mukundansundar) [inline pkg/runtime/cli.go:136]: should we warn as this might lead to some confusion for users using Self Hosted mode if the AppID changes without info.

Comment (shubham1172) [reply pkg/runtime/cli.go:136]: Yes, we can do that. Let me add it.

**Review by ItalyPaleAle — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Comment by ItalyPaleAle** [comment]
> We should add new E2E test scenarios to make sure things break on breaking changes.

We do not have E2E scenarios for self-hosted currently :(

**Comment by yaron2** [comment]
> > We should add new E2E test scenarios to make sure things break on breaking changes.
> 
> We do not have E2E scenarios for self-hosted currently :(

We don't, but that's a point in time situation. It actually would be easier to orchestrate self hosted tests than our k8s e2e tests. I'm just wondering if the kind of test to catch something like this should actually be done on the components side or on the daprd side.

**Pull request merged**